fs-repo-11-to-12/migration/setup.go

```diff
@@ -83,6 +89,58 @@ func (m *Migration) open(opts migrate.Options) error {
 	return nil
 }
 
+func getUnexportedField(field reflect.Value) interface{} {
+	return reflect.NewAt(field.Type(), unsafe.Pointer(field.UnsafeAddr())).Elem().Interface()
+}
+
+func IsBasicFlatFSBlockstore(dstore ds.Datastore) (dsPath string, v1 *flatfs.ShardIdV1, err error) {
+	errNotDefault := errors.New("not the default config")
+	defer func() {
+		if err := recover(); err != nil {
+			err = errNotDefault
+		}
+	}()
+
+	mds, ok := dstore.(*mount.Datastore)
+	if !ok {
+		return "", nil, errNotDefault
+	}
+
+	mnts, ok := getUnexportedField(reflect.ValueOf(mds).Elem().FieldByName("mounts")).([]mount.Mount)
+	if !ok {
+		return "", nil, errNotDefault
+	}
+
+	if len(mnts) != 2 {
+		return "", nil, errNotDefault
+	}
+
+	var blkDs ds.Datastore
+	if mnts[0].Prefix.Equal(blocksPrefix) {
```

Comment:
Should I be checking that the other mount is for `/` here?

Reply:
It doesn't seem necessary to me given that we're checking for prefix equality (e.g. people shouldn't be able to mix things up if they've divided up the blockstore across multiple datastores)